### PR TITLE
Small Enchancements + VTune - Main Loop 1 of N

### PR DESCRIPTION
**Pull request by fmatthew5876**
I pulled some of the standalone commits out of #2017 to here. 

This adds some small improvements and includes the VTune integration. I made an attempt to put vtune frame instrumentation around our current main loop. It shows the CPU burn effect that gets fixed in #2017 

This is tested, ready for review and merge.

### CODE REVIEW

**Comment by Ghabry** [comment]
For all this main loop stuff pls don't rush it. Want some cooperation with emscripten asynchify here so we can finally get this in.
(Will update the buildscript 2020)

**Comment by fmatthew5876** [comment]
I fixed up the vtune cmake file. It also works on windows now if you specify `-DVTUNE_ROOT=<VTUNE INSTALL PATH`

File: resources/easyrpg-player.6.adoc [hunk 1]
```diff
@@ -50,6 +50,9 @@ for some additional binary patches.
 *--show-fps*::
   Enable frames per second counter.
 
+*--fps-render-window*::
+  Render the frames per second counter both full screen and windowed mode.
```
Comment (Ghabry) [inline resources/easyrpg-player.6.adoc]: counter _in_ both

Comment (fmatthew5876) [reply resources/easyrpg-player.6.adoc]: fixed

**Comment by fmatthew5876** [comment]
> 
> 
> Well, I'm currently not seeing the big picture of this instrumentation stuff. Looking forward to the next PRs to make this look nicer :P

The Framerate histogram in vtune isn't working for me in this version. But it does later after the main loop gets restructured.

Addressed review comments and rebased

File: src/scene.h [hunk 2]
```diff
@@ -198,7 +198,7 @@ class Scene {
 	static const char scene_names[SceneMax][12];
 
 	/** Called by the graphic system to request drawing of a background, usually a system color background */
-	virtual void DrawBackground();
+	virtual void DrawBackground(Bitmap& dst);
```
Comment (carstene1ns) [inline src/scene.h]: dox?

**Comment by fmatthew5876** [comment]
Rebased and fixed comments

**Pull request merged**